brute_force_detector/host.h

```diff
@@ -55,168 +55,175 @@
-
     }
 
     virtual ur_time_t getHostReportTimeout() = 0;
 
-    virtual bool canReportAgain(ur_time_t actualTime)
-    {
+    virtual bool canReportAgain(ur_time_t actualTime) {
         ur_time_t timer = getHostReportTimeout();
         return checkForTimeout(timeOfLastReport, timer, actualTime);
     }
 
     virtual ur_time_t getHostAttackTimeout() = 0;
 
-    virtual bool checkForAttackTimeout(ur_time_t actualTime)
-    {
+    virtual bool checkForAttackTimeout(ur_time_t actualTime) {
         ur_time_t timer = getHostAttackTimeout();
         return checkForTimeout(timeOfLastReport, timer, actualTime);
     }
 
     virtual ATTACK_STATE checkForAttack(ur_time_t actualTime) = 0;
 
-    RecordList<T>* getPointerToIncomingRecordList() {return &recordListIncoming; }
-    RecordList<T>* getPointerToOutgoingRecordList() {return &recordListOutgoing; }
+    RecordList<T> *getPointerToIncomingRecordList() { return &recordListIncoming; }
+
+    RecordList<T> *getPointerToOutgoingRecordList() { return &recordListOutgoing; }
 
-    virtual void clearAllRecords() { recordListIncoming.clearAllRecords(); recordListOutgoing.clearAllRecords();}
+    virtual void clearAllRecords() {
+        recordListIncoming.clearAllRecords();
+        recordListOutgoing.clearAllRecords();
+    }
 
-    bool isFlowScan(uint32_t *packets, uint8_t *flags)
-    {
-        if((*packets == 1 && *flags == 0b00000010) //SYN
-                || (*packets == 2 && *flags == 0b00000010) //SYN
-                || (*packets == 2 && *flags == 0b00000110) //SYN + RST
-                || (*packets == 1 && *flags == 0b00010010) //SYN + ACK
-                || (*packets == 1 && *flags == 0b00010100) //RST + ACK
-                || (*packets == 3 && *flags == 0b00000010))//3-syn packets
+    bool isFlowScan(const uint32_t *packets, const uint8_t *flags) {
+        if ((*packets == 1 && *flags == 0b00000010)    // SYN
+            || (*packets == 1 && *flags == 0b00010010)  // SYN + ACK
+            || (*packets == 1 && *flags == 0b00010100)  // RST + ACK
+            || (*packets == 2 && *flags == 0b00000010)  // SYN
+            || (*packets == 2 && *flags == 0b00000110)  // SYN + RST
+            || (*packets == 3 && *flags == 0b00000010)) // 3 SYN packets
         {
             scanned = true;
             return true;
         }
-        else
+        else {
             return false;
+        }
     }
 
 protected:
-    bool checkForTimeout(ur_time_t flowTime, ur_time_t timer, ur_time_t actualTime)
-    {
-        if(flowTime + timer <= actualTime)
-            return true;
-        else
-            return false;
-    }
 
     bool scanned;
 
     ip_addr_t hostIp;
-    ur_time_t firstSeen;
+    ur_time_t firstSeen; // TODO unused
     ur_time_t timeOfLastReport;
     ur_time_t timeOfLastReceivedRecord;
-    RecordList<T> recordListIncoming; //incoming direction to victim (attacker -> victim)
+    RecordList<T> recordListIncoming; // direction to victim (attacker -> victim)
```

Comment:
It is a challenge for me to understand, why "incoming" represents traffic that is generated by attacker :-)